protoc-gen-grpc-gateway/internal/gengateway/generator.go

```diff
@@ -121,6 +124,24 @@ func (g *generator) Generate(targets []*descriptor.File) ([]*plugin.CodeGenerato
 	return files, nil
 }
 
+func (g *generator) getFilePath(file *descriptor.File) (string, error) {
+	name := file.GetName()
+	if g.modulePath != "" {
+		if g.pathType != pathTypeImport {
+			return "", errors.New("cannot use module= with paths=source_relative")
+		}
+		trimPath, pkgPath := g.modulePath+"/", file.GoPkg.Path+"/"
+		if !strings.HasPrefix(pkgPath, trimPath) {
+			return "", fmt.Errorf("%v: file go path does not match module prefix: %v", file.GoPkg.Path, trimPath)
+		}
+		name = filepath.Join(strings.TrimPrefix(pkgPath, trimPath), filepath.Base(name))
+	} else if g.pathType == pathTypeImport && file.GoPkg.Path != "" {
+		// Only respect the path value if module is not supplied
+		name = fmt.Sprintf("%s/%s", file.GoPkg.Path, filepath.Base(name))
+	}
```

Comment:
Whenever I see something like `else if` I think it can usually be done neater with a switch statement. What do you think about:
```suggestion
	switch {
	case g.modulePath != "" && g.pathType != pathTypeImport:
		return "", errors.New("cannot use module= with paths=source_relative")
	case g.modulePath != "":
		trimPath, pkgPath := g.modulePath+"/", file.GoPkg.Path+"/"
		if !strings.HasPrefix(pkgPath, trimPath) {
			return "", fmt.Errorf("%v: file go path does not match module prefix: %v", file.GoPkg.Path, trimPath)
		}
		name = filepath.Join(strings.TrimPrefix(pkgPath, trimPath), filepath.Base(name))
	case g.pathType == pathTypeImport && file.GoPkg.Path != "":
		// Only respect the path value if module is not supplied
		name = fmt.Sprintf("%s/%s", file.GoPkg.Path, filepath.Base(name))
	}
```